clang/test/CodeGenHLSL/resources/res-array-global-subarray-many.hlsl

```diff
@@ -0,0 +1,102 @@
+// RUN: %clang_cc1 -triple dxil-pc-shadermodel6.6-compute -finclude-default-header \
+// RUN:   -emit-llvm -disable-llvm-passes -o - %s | FileCheck %s
+
+// CHECK: @[[BufA:.*]] = private unnamed_addr constant [2 x i8] c"A\00", align 1
+
+RWBuffer<float> A[5][4][3][2] : register(u10, space2);
+RWStructuredBuffer<float> Out;
+
+float foo(RWBuffer<float> Arr[3][2]) {

```

Comment:
Not seeing any file check statements for the code generated by foo. Is that something we'd want to check to make sure the subscript accesses are in the correct order?

Reply:
I wanted the focus of this test to be just on the indexing of subarrays. Testing of local resource array function argument is covered in `res-array-local2.hlsl` that was added in https://github.com/llvm/llvm-project/pull/153257. But I guess it does not hurt to add a few check lines here as well to test the multi-dimensional local array.